components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacetImpl.java

```diff
@@ -72,6 +75,32 @@ public SearchFacetImpl(final SearchService searchService,
   {
     this.searchService = checkNotNull(searchService);
     this.componentMetadataProducers = checkNotNull(componentMetadataProducers);
+    this.rebuildIndexLock = new ReentrantReadWriteLock();
+  }
+
+  @Override
+  @Guarded(by = STARTED)
+  public void rebuildIndex() {
+    rebuildIndexLock.writeLock().lock();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) org.sonatype.nexus.repository.search.SearchFacetImpl.rebuildIndex() does not release lock on all exception paths [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3AUL_UNRELEASED_LOCK_EXCEPTION_PATH)